### DISCUSSION TITLE
prerending-a-blazor-webassembly-app-without-an-asp-net-core-host-app/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#110] [Announcements]
# prerending-a-blazor-webassembly-app-without-an-asp-net-core-host-app/

In this post I show how to prerender all the pages in a Blazor WebAssembly app ahead of time, so you can host the app as static files, without a host app.

https://andrewlock.net/prerending-a-blazor-webassembly-app-without-an-asp-net-core-host-app/

**Comment by CalvinQuark** [comment]
Thanks Andrew! This post is quite helpful. However, I wonder if it is possible to ensure that only a pre-authorized static Blazor app (whether prerendered or not) is granted access to backend Azure Functions. If so, I have one use case where this approach is particularly apropos.

[a comment by an automated account is not shown]

**Comment by jaredthirsk** [comment]
Do the new features in .NET 8 offer any built-in ways to do some or all of this?
